app/reducers/rootReducer.ts

```diff
@@ -1,94 +1,140 @@
 import { POPUP_PROPS, Action } from "../types";
+import * as actionType from "../actions";
+
 export const INITIAL_POPUP_STATE: POPUP_PROPS = {
-  enabled: false,
-  tabId: -1,
-  tabUrl: "",
-  errorMessage: "",
+  data: {},
+  currentUrl: "",
+  currentTab: -1
+};
+
+const initialTabProperties = {
+  enabledStatus: false,
   requests: [],
+  interceptStatus: "",
+  errorMessage: "",
+  PageDetails: [],
   checkedReqs: {},
-  responseText: {},
-  statusCodes: {},
   contentType: {},
-  PageDetails: [],
-  interceptStatus: "",
-  isInterceptorOn: {},
+  isInterceptorOn: true,
   responseData: {},
-  responseError: {}
+  responseError: {},
+  responseText: {},
+  statusCodes: {}
 };
 
-//ACTION CONSTANTS
-import * as actionType from "../actions";
+function extendStateData(state: POPUP_PROPS, payload: any, props: any) {
+  return {
+    ...state,
+    data: {
+      ...state.data,
+      [payload.tabId]: {
+        ...state.data[payload.tabId],
+        ...props
+      }
+    }
+  };
+}
 
+//ACTION CONSTANTS
 export const reducer = (state = INITIAL_POPUP_STATE, action: Action) => {
   switch (action.type) {
-    case actionType.START_LISTENING:
-      return { ...state, enabled: action.payload };
-    case actionType.UPDATE_FIELD:
-      return { ...state, [action.payload.name]: action.payload.value };
-    case actionType.UPDATE_FIELDS:
-      return { ...state, ...action.payload };
-    case actionType.STOP_LISTENING:
-      return { ...state, enabled: action.payload };
+    case actionType.INITIALISE_DEFAULTS:
+      return {
+        ...state,
+        data: {
+          ...state.data,
+          [action.payload.currentTab]: {
+            ...(state.data[action.payload.currentTab] || initialTabProperties)
+          }
+        },
+        currentUrl: action.payload.currentUrl,
+        currentTab: action.payload.currentTab
+      };
     case actionType.ERROR:
-      return { ...state, errorMessage: action.errorMessage, enabled: false };
+      return extendStateData(state, action.payload, {
+        errorMessage: action.payload.errorMessage,
+        enabledStatus: false
+      });
     case actionType.CLEAR_REQUESTS:
-      return { ...state, requests: [] };
+      return extendStateData(state, action.payload, {
+        requests: []
+      });
     case actionType.TOGGLE_CHECKBOX:
-      return {
-        ...state,
+      return extendStateData(state, action.payload, {
         checkedReqs: {
-          ...state.checkedReqs,
+          ...state.data[action.payload.tabId].checkedReqs,
```

Comment:
Why do we need to pass this again frmo here. This is already doing in `extendStateData`.